packages/ovee/src/utils/Task.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+type Resolve<T> = (value: T | PromiseLike<T>) => void;
+type Reject = (reason?: any) => void;
+
+export class Task<T = void> extends Promise<T> {
```

Comment:
Maybe we could move the task scheduling into this class? So that the use of `requestAnimationFrame` is a responsibility of Task, not the Component itself?

Reply:
I would like to keep `Task` as a utility tool but we can create similar, like `ScheduleTask` or smth similar

Reply:
Would you do the refactor for 2.1 or we push it to future version?

Reply:
I think it's the same as upper comment and this issue: #17 :D

Reply:
Okay, 2.2 candidate then